app/frontend/controllers/event_controller.js

```diff
@@ -0,0 +1,136 @@
+/* This file is part of Ezra Bible App.
+
+   Copyright (C) 2019 - 2021 Ezra Bible App Development Team <contact@ezrabibleapp.net>
+
+   Ezra Bible App is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 2 of the License, or
+   (at your option) any later version.
+
+   Ezra Bible App is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with Ezra Bible App. See the file LICENSE.
+   If not, see <http://www.gnu.org/licenses/>. */
+
+
+/**
+ * This controller implements pub/sub pattern and serves as a broker (event bus) between
+ * event emitter (publisher) and event consumer subscriber.
+ * https://en.wikipedia.org/wiki/Publish%E2%80%93subscribe_pattern
+ * 
+ * @module eventController
+ * @category Controller
+ */
+
+var messages = {};
+
+function notCreated(event) {
+  return !(event in messages);
+}
+
+/**
+ * @typedef { "on-start-repo-update" | "on-progress-repo-update" | "on-complete-repo-update" } RepoUpdateEvents
+ */
+/**
+ * For consistency event name should be a string in-kebab-case with the following template: "on-[action]-[object]"
+ * @typedef { RepoUpdateEvents } Event
+ */
```

Comment:
I would suggest to move these typedefs to the place where these events are published.

Otherwise we have to keep touching this controller whenever we add events.

Reply:
I get it now based on the auto completion I have seen while working on it. However, would it make sense to move all event typedefs to a dedicated file? (like events.js) Or would that then mean that all subscribers would need to require two modules instead of just one?